source/connect/tls.txt

```diff
@@ -0,0 +1,241 @@
+.. c-tls:
+
+========================================
+Configure Transport Layer Security (TLS) 
+========================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: security, authentication, transport layer security, encrypt
+
+Overview
+--------
+
+In this guide, you can learn how to use the :wikipedia:`TLS <Transport_Layer_Security>`
+protocol to secure your connection to a MongoDB deployment.
+
+When you enable TLS for a connection, the {+driver-short+} performs the following actions:
+
+- Uses TLS to connect to the MongoDB deployment
+- Verifies the deployment's certificate
+- Ensures that the certificate certifies the deployment
+
+To learn how to configure your MongoDB deployment for TLS, see the
+:manual:`TLS configuration guide </tutorial/configure-ssl/>` in the
+MongoDB Server manual. 
+
+.. important::
+
+   A full description of TLS/SSL, PKI (Public Key Infrastructure) certificates, and
+   Certificate Authorities (CAs) is beyond the scope of this document. This page assumes prior
+   knowledge of TLS/SSL and access to valid certificates.
+
+.. _c-enable-tls:
+
+Enable TLS
+----------
+
+You can enable TLS on a connection to your MongoDB instance
+in the following ways:
+
+- Setting the ``tls`` parameter in your connection string
+- Using the ``mongoc_uri_set_option_as_bool()`` function to set the ``MONGOC_URI_TLS`` connection
+  option to ``true``
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: connectionstring
+
+      .. literalinclude:: /includes/connect/tls-connection-str.c
+         :language: c
+         :start-after: start-connect-to-atlas
+         :end-before: end-connect-to-atlas
+         :dedent:
+
+   .. tab:: MongoC URI Options
+      :tabid: mongocurioptions
+
+      .. literalinclude:: /includes/connect/tls-uri-options.c
+         :language: c
+         :start-after: start-connect
+         :end-before: end-connect
+         :dedent:
+
+.. tip::
+   
+   If your connection string includes the ``+srv`` modification, which specifies the
+   SRV connection format, TLS is enabled on your connection by default.
+
+   To learn more about the SRV connection format, see
+   :manual:`SRV Connection Format </reference/connection-string/#srv-connection-format>`
+   in the {+mdb-server+} documentation.
+
+.. _c-specify-ca-file:
+
+Specify a CA File
+------------------
+
+When MongoDB is started with TLS enabled, it will by default require the client to provide
+a client certificate issued by a certificate authority specified by `--tlsCAFile`, or an authority
+trusted by the native certificate store in use on the server.
+
+To provide the client certificate, set the `tlsCertificateKeyFile` in the URI to a PEM armored
```

Comment:
S: since the code example is using the `mongoc_uri_set_option_as_utf8()` function to set the `MONGOC_URI_TLSCERTIFICATEKEYFILE` option, rather than including `tlsCertificateKeyFile` in the URI, I'd adjust the wording to reflect that:
```suggestion
To provide the client certificate, use the `mongoc_uri_set_option_as_utf8()` function to set the MONGOC_URI_TLSCERTIFICATEKEYFILE option to to a `.pem` file containing the root certificate chain:
```

Reply:
Is there also a way to set this option directly in the URI?